commonknowledge/wagtail/embed_providers/shiny.py

```diff
@@ -0,0 +1,56 @@
+
+import re
+from html.parser import HTMLParser
+from urllib.parse import urlparse
+from wagtail.embeds.finders.base import EmbedFinder
+import requests
+from django.utils.text import slugify
+
+
+class ShinyFinder(EmbedFinder):
+    SHINY_URL_PATTERN = re.compile(
+        r'https://stop-watch.shinyapps.io/(.+)/?')
+
+    def accept(self, url):
+        """
+        Returns True if this finder knows how to fetch an embed for the URL.
+        This should not have any side effects (no requests to external servers)
+        """
+        return self.SHINY_URL_PATTERN.match(url) is not None
+
+    def find_embed(self, url, max_width=None):
+        """
+        Takes a URL and max width and returns a dictionary of information about the
+        content to be used for embedding it on the site.
+
+        This is the part that may make requests to external APIs.
+        """
+
+        return {
+            'title': "Title of the content",
```

Comment:
One could use a modified `_ThumbnailExtract` class to get the page title.

But this isn't actually useful to end users, it's just a database thing, so no stress, no changes required.